test/lib/pgBackRestTest/Module/Common/CommonIoBufferedTest.pm

```diff
@@ -0,0 +1,233 @@
+        $self->testResult(sub {$oIoBuffered->readLine()}, '12345678', 'read 8 char line');
+        $oIoBuffered->writeLine();
+
+        #---------------------------------------------------------------------------------------------------------------------------
+        $self->testResult(sub {$oIoBuffered->readLine()}, '1', 'read 1 char line');
+        $self->testResult(sub {$oIoBuffered->readLine()}, '2', 'read 1 char line');
+        $self->testResult(sub {$oIoBuffered->readLine()}, '345678', 'read 6 char line');
+        $oIoBuffered->writeLine();
+
+        #---------------------------------------------------------------------------------------------------------------------------
+        $self->testResult(sub {$oIoBuffered->readLine()}, '', 'read blank line');
+        $oIoBuffered->writeLine();
+
+        #---------------------------------------------------------------------------------------------------------------------------
+        $self->testException(sub {$oIoBuffered->readLine()}, ERROR_FILE_READ, 'unexpected EOF reading line from socket client');
+        $self->testException(
+            sub {$oIoBuffered->readLine(false)}, ERROR_FILE_READ, 'unexpected EOF reading line from socket client');
+        $self->testResult(sub {$oIoBuffered->readLine(true)}, undef, 'ignore EOF');
+    }
+
+    ################################################################################################################################
+    if ($self->begin('read'))
+    {
+        my $tBuffer;
+
+        my $oClient = $self->socketServer($strSocketFile, sub
+        {
+            my $oIoHandle = shift;
+
+            my $tResponse = '';
+            my $tData;
+
+            # Ack after timeout
+            while (length($tResponse) != 1) {$oIoHandle->read(\$tResponse, 1)};
+
+            # Write mixed buffer
+            $tResponse = '';
+            $tData = "123\n123\n1";
+            $oIoHandle->write(\$tData);
+            while (length($tResponse) != 1) {$oIoHandle->read(\$tResponse, 1)};
+
+            # Write buffer
+            $tResponse = '';
+            $tData = "23456789";
+            $oIoHandle->write(\$tData);
+
+            # Wait before writing the rest to force client to loop
+            usleep(500);
+            $tData = "0";
+            $oIoHandle->write(\$tData);
+            while (length($tResponse) != 1) {$oIoHandle->read(\$tResponse, 1)};
+        });
+
+        #---------------------------------------------------------------------------------------------------------------------------
+        my $oIoBuffered = $self->testResult(
+            sub {new pgBackRest::Common::Io::Buffered(
+                new pgBackRest::Common::Io::Handle('socket client', $oClient, $oClient), 1, 8)}, '[object]', 'open');
+
+        $self->testException(
+            sub {$oIoBuffered->read(\$tBuffer, 1, true)}, ERROR_FILE_READ,
+            'unable to read 1 byte(s) after 1 second(s) from socket client');
+
```

Comment:
readLine() doesn't take a buffer as an argument, so not quite sure what you are trying to do here.

I did get the module to 100% coverage, though.